open_learning_ai_tutor/intent_selector.py

```diff
@@ -0,0 +1,81 @@
+from open_learning_ai_tutor.constants import Intent
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def extract_assessment_codes(assessment):
+    try:
+        json_data = json.loads(assessment)
+        selection = json_data["selection"]
+        codes = list(selection)
+        return codes
+    except json.decoder.JSONDecodeError:
+        logger.error(
+            f"Error decoding assessment: {assessment}. Defaulting to m) The student is asking about concepts"
+            "or information related to the materialcovered by the problem, or is continuing such a discussion. "
+        )
+        return ["m"]
+
+
+def get_intent(assessment, previous_intent):
+    assessment_codes = extract_assessment_codes(assessment)
+
+    intents = []
+    if "l" in assessment_codes:
+        return [Intent.G_REFUSE]
+
+    if "m" in assessment_codes:
+        intents.append(Intent.S_STATE)
+        intents.append(Intent.A_CURIOSITY)
+
+    if "i" in assessment_codes:
+        intents.append(Intent.S_CALCULATION)
+
+    if "h" in assessment_codes:
+        intents.append(Intent.S_STATE)
+
+    if "e" in assessment_codes and "g" not in assessment_codes:
+        intents.append(Intent.P_ARTICULATION)
+
+    if Intent.S_SELFCORRECTION in previous_intent:
+        if "f" in assessment_codes:
+            intents.append(Intent.S_STRATEGY)
+        elif "a" in assessment_codes or "b" in assessment_codes:
+            intents.append(Intent.S_CORRECTION)
+        if "c" in assessment_codes:
+            intents.append(Intent.S_CALCULATION)
+
+    if "g" in assessment_codes:
+        if Intent.P_HYPOTHESIS in previous_intent:
+            intents.append(Intent.S_HINT)
+        elif "d" not in assessment_codes and "f" not in assessment_codes:
+            intents.append(Intent.P_HYPOTHESIS)
+
+        if Intent.S_CORRECTION in previous_intent:
+            intents.append(Intent.S_CORRECTION)
+
+    if (
+        any(code in assessment_codes for code in ("a", "b", "c"))
+        and not Intent.S_CALCULATION in intents
+    ):
+        intents.append(Intent.S_SELFCORRECTION)
+    elif "d" in assessment_codes:
+        intents.append(Intent.S_STRATEGY)
+        intents.append(Intent.S_HINT)
+
+    if (
+        not (Intent.S_CORRECTION in intents)
+        and not (Intent.S_SELFCORRECTION in intents)
+        and not (Intent.S_CALCULATION in intents)
```

Comment:
might be more clear to rewrite using 'not any' similar to [here](https://github.com/mitodl/open-learning-ai-tutor/pull/5/files#diff-c31691ee52999bc511296b9726ac2a9bc68d157706ee1b25816c9425694787eeR60)